script.js

```diff
@@ -0,0 +1,141 @@
+let template = document.querySelector('#Video_template').content;
+
+document.querySelector("#search-button").addEventListener("click", function () {
+    if (document.querySelector("#search-query").value == "") {
+        alert("Please enter a search term");
+        return
+    }
+    document.querySelector(".main").innerHTML = "";
+    fetch(`https://www.googleapis.com/youtube/v3/search?key=AIzaSyB0Yem1_WTQl_SDGS1Sxkb46EnEKQiV5vI&type=video&part=snippet&maxResults=15&q=${document.querySelector("#search-query").value}`)
+        .then(searchResults => searchResults.json())
+        .then(searchResults => {
+            let ids = []
+            for (let video of searchResults.items) {
+                ids.push(video.id.videoId);
+            }
+            fetch(`https://www.googleapis.com/youtube/v3/videos?key=AIzaSyB0Yem1_WTQl_SDGS1Sxkb46EnEKQiV5vI&id=${ids.join(',')}&part=snippet,statistics`)
+                .then(videoDetails => videoDetails.json())
+                .then(videoDetails => {
+                    for (let i in searchResults.items)
+                        videoDetails.items[i].snippet.shortDesc = searchResults.items[i].snippet.description;
+                    getVid(videoDetails.items)
+                });
+        });
+//     fetch("./search.json")
+//         .then(searchResults => searchResults.json())
+//         .then(searchResults => {
+//             let ids = []
+//             for (let video of searchResults.items) {
+//                 ids.push(video.id.videoId);
+//             }
+//             fetch("./searchvide.json")
+//                 .then(videoDetails => videoDetails.json())
+//                 .then(videoDetails => {
+//                     for (let i in searchResults.items)
+//                         videoDetails.items[i].snippet.shortDesc = searchResults.items[i].snippet.description;
+//                     getVid(videoDetails.items)
+//                 });
+//         });
+ });
+
+
+
+
+
+
+function getVid(videoDetails) {
+    let i = 1
+    for (let video of videoDetails) {
+        fillDetails(video, i++);
+    }
+    hideAll();
+    displayVideos(1)
```

Comment:
please don't put two separate functions into one. it will violate the SRP principle.
As per the function name get video should returns the video metadata and display video / play video should display the video on screen

Reply:
resolved